src/sentry/uptime/migrations/0040_uptime_backfill_detector_grouphash.py

```diff
@@ -0,0 +1,100 @@
+
+
+def update_auto_detected_active_interval_seconds(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    GroupHash = apps.get_model("sentry", "GroupHash")
+    DataSource = apps.get_model("workflow_engine", "DataSource")
+    ProjectUptimeSubscription = apps.get_model("uptime", "ProjectUptimeSubscription")
+
+    for data_source in RangeQuerySetWrapperWithProgressBar(
+        DataSource.objects.filter(type=DATA_SOURCE_UPTIME_SUBSCRIPTION)
+    ):
+        uptime_subscription_id = int(data_source.source_id)
+        project_subscription_id = ProjectUptimeSubscription.objects.get(
+            uptime_subscription_id=uptime_subscription_id
+        ).id
+        detector = data_source.detectors.first()
+
+        detector_fingerprint_hash = hash_fingerprint(build_detector_fingerprint_component(detector))
+        sub_fingerprint_hash = hash_fingerprint(
+            build_subscription_fingerprint_component(project_subscription_id)
+        )
+        group_hashes = list(
+            GroupHash.objects.filter(
+                project=detector.project,
+                hash__in=[detector_fingerprint_hash, sub_fingerprint_hash],
+            )
+        )
+        # If we have 0 or 2 group hashes then we don't need to backfill anything. Either the group doesn't exist
+        # at all for this detector, or both hashes are already associated
+        if len(group_hashes) != 1:
+            continue
+        group_hash = group_hashes[0]
+        # If the only hash that exists is for the detector fingerprint then we don't have any work to do
+        if group_hash.hash == detector_fingerprint_hash:
+            continue
+
+        GroupHash.objects.get_or_create(
+            project=detector.project,
+            hash=detector_fingerprint_hash,
+            defaults={
+                "group": group_hash.group,
+            },
+        )
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = True
+
+    dependencies = [
```

Comment:
I think you’ll need to depend on the workflow engine migrations too. Otherwise the app state might not be right when this migration runs?

Reply:
I think that previous uptime migrations are depending on this, so it's probably fine. But added a dependency on the latest anyway

Reply:
Oh yeah you’re right. But I think it’s probably worth it to be specific? Not sure lol